go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -180,44 +191,74 @@ func (ef *egressFirewall) addACLToJoinSwitch(joinSwitches []string, hashedAddres
 			action = "drop"
 		}
 
-		ipAddress, _, err := net.ParseCIDR(rule.to.cidrSelector)
-		if err != nil {
-			return fmt.Errorf("error rule.to.cidrSelector %s is not a valid CIDR (%+v)", rule.to.cidrSelector, err)
-		}
-		if !utilnet.IsIPv6(ipAddress) {
-			match = fmt.Sprintf("match=\"ip4.dst == %s && ip4.src == $%s\"", rule.to.cidrSelector, hashedAddressSetNameIPv4)
+		if rule.to.cidrSelector != "" {
+			ipAddress, _, err := net.ParseCIDR(rule.to.cidrSelector)
+			if err != nil {
+				return fmt.Errorf("error rule.to.cidrSelector %s is not a valid CIDR (%+v)", rule.to.cidrSelector, err)
+			}
+			if !utilnet.IsIPv6(ipAddress) {
+				match = fmt.Sprintf("match=\"ip4.dst == %s && ip4.src == $%s\"", rule.to.cidrSelector, hashedAddressSetNameIPv4)
+			} else {
+				match = fmt.Sprintf("match=\"ip6.dst == %s && ip6.src == $%s\"", rule.to.cidrSelector, hashedAddressSetNameIPv6)
+			}
+
+			if len(rule.ports) > 0 {
+				match = fmt.Sprintf("%s && ( %s )\"", match[:len(match)-1], egressGetL4Match(rule.ports))
+			}
+			for _, joinSwitch := range joinSwitches {
+				err = createACLRule(defaultStartPriority-rule.id, match, action, ef.namespace, joinSwitch)
+				if err != nil {
+					return err
+				}
+			}
 		} else {
-			match = fmt.Sprintf("match=\"ip6.dst == %s && ip6.src == $%s\"", rule.to.cidrSelector, hashedAddressSetNameIPv6)
+			// rule based on DNS NAME
+			if rule.access == egressfirewallapi.EgressFirewallRuleDeny {
+				return fmt.Errorf("do not use dnsName as part of a deny rule this is insecure")
+			}
+			var portMatch string
+			if oc.egressFirewallDNS == nil {
+				var err error
+				oc.egressFirewallDNS, err = NewEgressDNS()
+				if err != nil {
+					return err
+				}
+				oc.egressFirewallStopChan = make(chan struct{})
+				go utilwait.Until(oc.egressFirewallDNS.Sync, 0, oc.egressFirewallStopChan)
```

Comment:
Will this spawn a goroutine every time an egress firewall is updated?

Reply:
@squeed, no it won't it spawns one goroutine the first time an egressFirewall rule with a dnsName is created (in any namespace) `Sync()` handles all namespaces. The beginning of this block starts with `if oc.egressFirewallDNS == nil` which will only happen when there is no egressFirewallDNS object on the controller

Reply:
Ah. That should also be pulled out of this function.

Reply:
pulled spawning the goroutine out